### ISSUE TITLE
Upgrade to Meteor 1.2.0.1 breaks the app

### ISSUE THREAD

**Issue by cellulosa** [Urigo/angular-meteor#703]
```javascript
[Error] Error: [$injector:nomod] Module 'myApp' is not available! You either misspelled the module name or forgot to load it. If registering a module ensure that you specify the dependencies as the second argument.
http://errors.angularjs.org/1.4.6/$injector/nomod?p0=myApp
    (anonymous function) (angular_angular.js, line 2019)
    ensure (angular_angular.js, line 1941)
    module (angular_angular.js, line 2015)
    (anonymous function) (routes.ng.js, line 1)
    global code (routes.ng.js, line 58)
```

The app is built following the [Getting started angular-meteor tutorial](https://angular-meteor.com/tutorialIntro).

```javascript
// client/lib/app.js
angular.module('myApp', ['angular-meteor', 'ui.router']);
```

```javascript
// client/routes.ng.js
angular.module("myApp").config(
    function($urlRouterProvider, $stateProvider, $locationProvider){

        $locationProvider.html5Mode(true);

        $stateProvider
            .state('index', {
                url: '/',
                templateUrl: 'client/index/views/index.ng.html',
                controller: 'IndexCtrl'
            });

        $urlRouterProvider.otherwise('/404');
    }
);
```

```javascript
// client/index/controllers/index.ng.js
angular.module("myApp").controller("IndexCtrl",
    function($scope, $meteor, $state){

    }
);
```

**Comment by veekungx** [comment]
You forgot "injection array"

angular.module("myApp")

to

angular.module("myApp",[])

**Comment by shantanubhadoria** [comment]
I am facing the same error. The whole load order seems to be broken. Even though my app.js is inside client/lib it still doesn't load before other files in client/

The new Meteor Package has lot of changes. I am not sure you will get the same result all by a straight upgrade of a existing meteor code as the default package names have changed. 
meteor-package is changed to meteor-base among other new addition. 
To replicate this error you will have to start a new meteor base project, do 'meteor add angular' and then go from there.
HTH

**Comment by shantanubhadoria** [comment]
Correction the update does replace the meteor-package with meteor-base as well. I just upgraded my pre-1.2.0 app and face the same error. I think this has something to do with load error. Because as soon as I upgrade to 1.2.0 the client/lib js files don't load before the other ng.js files anymore. Downgrading restores the load order for the files.
Hope that helps

**Comment by shantanubhadoria** [comment]
Ok I just figured out what breaks my application. In the older version my app.js routes.js and directives.js files were loaded before all the ng.js controller files because they were in client/lib/ folder. In the newer meteor version, all ng.js files get loaded before everything else inspite of the other js files being in lib/ folder. Once I rename my app.js, routes.js, directives.js to app.ng.js, routes.ng.js, directives.ng.js Everything works well and the load order is restored. I think this is not a problem as lng as this change is documented.

Cheers!

**Comment by Urigo** [comment]
@shantanubhadoria that is interesting...   so `.ng.js` files are loaded before `.js` no matter what?

**Comment by shantanubhadoria** [comment]
Yep, That seems to be the case. I can say atleast client/lib/*.js gets loaded after client/abc/*.ng.js

**Comment by netanelgilad** [comment]
Well I made a quick reproduction and opened an issue to the guys at meteor here https://github.com/meteor/meteor/issues/5285. For the mean time, we can document that somewhere...

**Comment by glasser** [comment]
Hey folks, see my comment over there:  it looks like the angular plugin had code in it that triggers an obscure feature of Meteor designed for exactly this purpose (putting template files before everything else), but due to a bug in old Meteor that line was being ignored. Just remove the line and it should be good!

**Comment by Urigo** [comment]
Fixed in `1.0.0-rc.10`

**Comment by vacarsu** [comment]
@Urigo I seem to be having this issue in Cordova on android-device. All of my .ng.js files are loading before my /lib folder. I am using RC10.
